src/sound/manager.cpp

```diff
@@ -0,0 +1,206 @@
+	if (findInVector(v_clips,sc) != v_clips.end()) {
+		std::cerr << ("SoundManager -> addClip: SoundClip already exist\n");
+		return;
+	}
+
+	v_clips.push_back(sc);
+}
+
+void SoundManager::removeClip(std::shared_ptr<SoundClip> sc) {
+	removeFromVector(v_clips, sc);
+}
+
+void SoundManager::addGroup(std::shared_ptr <SoundGroup> sg) {
+	if (!sg) {
+		std::cerr << ("SoundManager -> addGroup: SoundGroup not exist\n");
+		return;
+	}
+
+	removeExpired(v_groups);
+	// check if exist a source with a given name
+	if (findInVector(v_groups, sg) != v_groups.end()) {
+		std::cerr << ("SoundManager -> addGroup: SoundGroup already exist\n");
+		return;
+	}
+	v_groups.push_back(sg);
+}
+
+void SoundManager::loadAudioToClip(std::shared_ptr <SoundClip> sc, const char* path) {
+	if (!sc) {
+		std::cerr << ("SoundManager -> addAudioToClip: Shared pointer SoundClip sc is nullptr\n");
+		return;
+	}
+	
+	auto clip_find = findInVector(v_clips, sc);
+	// check if exist a clip with a given name
+	if (clip_find == v_clips.end()) {
+		std::cerr << ("SoundManager -> addAudioToClip: SoundClip not already exist\n");
+		return;
+	}
+	
+	sc->loadAudio(path);
+}
+
+void SoundManager::connectClipWithSource(std::shared_ptr<SoundClip> sc, std::shared_ptr <SoundSourceObject> sso) {
+	if (!sso) {
+		std::cerr << ("SoundManager -> connectClipWithSource: SoundSourceObject not exist\n");
+		return;
+	}
+
+	if (!sc) {
+		std::cerr << ("SoundManager -> connectClipWithSource: SoundClip not exist\n");
+		return;
+	}
+
+	removeExpired(v_sources);
+	auto source_find = findInVector(v_sources, sso);
+	auto clip_find = findInVector(v_clips, sc);
+
+	//@TODO Maybe intead of an exception we should create a Clip or SourceObjecy
+
+	// check if exist a clip with a given name
+	if (clip_find == v_clips.end()) {
+		std::cerr << ("SoundManager -> connectClipWithSource: SoundClip not already exist\n");
+		return;
+	}
+
+	// check if exist a source with a given name
+	if (source_find == v_sources.end()) {
+		std::cerr << ("SoundManager -> connectClipWithSource: SoundSourceObject not already exist in v_sources. Please add to map (sm.addSource(sso))\n");
+		return;
+	}
+
+	sso->addBuffer(sc);
+}
+
+std::shared_ptr<SoundClip> SoundManager::createSoundClipAndLoadAudio(const char* path) {
+	auto sc = std::make_shared<SoundClip>();
+	addClip(sc);
+	loadAudioToClip(sc, path);
+	return sc;
+}
+
+std::shared_ptr<SoundClip> SoundManager::createSoundClipAndAddToSourceObject(const char* path, std::shared_ptr<SoundSourceObject> sso) {
+	auto sc = std::make_shared<SoundClip>();
+	addClip(sc);
+	loadAudioToClip(sc, path);
+	connectClipWithSource(sc, sso);
+	return sc;
+}
+
+void SoundManager::playSound(std::weak_ptr <SoundSourceObject> sso){
+	//std::lock_guard<std::mutex> lock(sound_manager_mutex);
+
+	if (sso.expired()) {
+		std::cerr << ("SoundManager -> playSound: SoundSourceObject not exist\n");
+		return;
+	}
+	
+	removeExpired(v_sources);
+	auto source_find = findInVector(v_sources, sso);

```

Comment:
What is the purpose of those `findInVector` calls? I don't really understand what they are checking for. Such an expensive debug-only operation is really unadvised